compare/compare.go

```diff
@@ -15,7 +16,20 @@ type CompareParams struct {
 	DisallowExtraFields  bool
 }
 
-// Compare ...
+type leafsMatchType int
+
+const (
+	pure leafsMatchType = iota
```

Comment:
Looks great to me. 

The only concern I have is regarding meaning of `pure`. It looks ambiguous to me. What about calling it `equals` or `simple`?

Reply:
I'm not sure.
`equal` is definitely worse, because it is not clear what we are comparing - map, array or simple text (all of them can be equal in its own way).
`simple` is better than `equal`, but for me, it is associated with simple nodes, i.e. leafs. And before I introduced the term `leaf ` here, `simple` was used this way.

So, how it looks like in my head:
- `equal` - when we compare two general structures (text, array, map)
- `simple` - when we compare leafs (i.e. text, not array or map). But it's not specified, _how_ we do this (it can be regex or plain/pure text)
- `pure` - when we compare text just as plain text, without any rules or expressions.

I don't argue here, I'm just trying to tell how do I take it. If it looks obviously only for me, it's definitely bad way.

Reply:
IMHO, pure is OK too. @What-If-I take `pure` as short for `pure strings`, while `regex` for `regular expressions`. )

Reply:
OK, probably I just have different associations with `pure` and never seen it's usage in this context. Got the point, thought. Pure seems reasonable.